stdcommands/mentionrole/mentionrole.go

```diff
@@ -74,27 +76,52 @@ func cmdFuncMentionRole(data *dcmd.Data) (interface{}, error) {
 	} else if !ok {
 		return "You need manage server perms to use this command", nil
 	}
-
+	
+	roleS := data.Args[0].Str()
 	var role *discordgo.Role
 	data.GS.RLock()
 	defer data.GS.RUnlock()
 	for _, r := range data.GS.Guild.Roles {
-		if strings.EqualFold(r.Name, data.Args[0].Str()) {
+		if strings.EqualFold(r.Name, roleS) {
 			role = r
 			break
 		}
 	}
+	
+	//if we did not find a match yet try to match ID
+	if role == nil {
+		parsedNumber, parseErr := strconv.ParseInt(roleS, 10, 64)
+		if parseErr == nil {
+			// was a number, try looking by id
+			role = data.GS.RoleCopy(false, parsedNumber)
+		}
+	}
 
 	if role == nil {
-		return "No role with the name `" + data.Args[0].Str() + "` found", nil
+		return "No role with the name or ID`" + roleS + "` found", nil
+	}
+	
+	cID := data.CS.ID
+	c := data.Switch("channel")
+	if c.Value != nil {
+		cID = c.Value.(*dstate.ChannelState).ID
+		
+		perms, err := data.GS.MemberPermissions(true, cID, data.Msg.Author.ID)
```

Comment:
set the first argument to false since were holding a read lock on the state already

Reply:
I see. Changed lock to false .